File: examples/01_TTS/python/tts_watson_ssml.py
# Example TTS - "watson with ssml"
# -*- coding: utf-8 -*-
import os
import sys
import urllib

def say(something):
    APIkey = "YOUR_API_KEY"
    url = "YOUR_API_URL"
    data = urllib.quote_plus(something)
    voice = "voice=en-US_AllisonVoice"
    command = "curl -X GET -u \"apikey:{0}\" --output say.mp3 \"{1}/v1/synthesize?accept=audio/mp3&text=\"{2}\"&{3}\" && afplay say.mp3 ".format(APIkey,url,data,voice)
    os.system(command)

def formatSSML(data):
    quoteFree = data.replace('"','\"')
    singleLine = quoteFree.replace('\r\n', '').replace('\n', '').replace('\r', '').replace('\t', '')
    return singleLine

def loadSSML(filename):
    file = open(filename,'r')
    data = file.read()
    say(formatSSML(data))


loadSSML('watson.xml')

File: examples/01_TTS/node/tts_watson_ssml.js
// Example TTS - "watson with ssml"

var exec = require('child_process').execSync;
var util = require('util');
var fs = require('fs');

function say(something){
  var APIkey = "YOUR_API_KEY";
  var url = "YOUR_API_URL";
  var data = encodeURIComponent(something);
  var voice = "voice=en-US_AllisonVoice";
  command = util.format('curl -X GET -u \"apikey:%s\" --output say.mp3 \"%s/v1/synthesize?accept=audio/mp3&text=\"%s\"&%s\" && afplay say.mp3 ', APIkey,url,data,voice);
  exec(command);
}

function formatSSML(data){
  var quoteFree = data.replace(/["]+/g, '\"');
  var singleLine = quoteFree.replace(/(\r\n|\n|\r|\t)/gm,"");
  return singleLine;
}

function loadSSML(filename){
  fs.readFile(filename, 'utf8', function(err, data) {
    say(formatSSML(data));
  });
}


loadSSML('watson.xml');
